openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java

```diff
@@ -93,11 +93,8 @@ public void generate(OpenApiDocConfig config, SyntaxNodeAnalysisContext context,
                     updateOpenApiContext(context, serviceId, openApiDefinition, embed);
                 } else {
                     // generate open-api doc and update the context if the `contract` configuration is not available
-                    generateOpenApiDoc(config, context, location);
+                    generateOpenApiDoc(config, context, location, embed);
                 }
-            } else {
-                // generate open-api doc and update the context
-                generateOpenApiDoc(config, context, location);
```

Comment:
Is this case already covered?

Reply:
In this case we do not generate open-api doc for now (cos unless it is manually enabled usiing open-api annotation it should not generate). So, removed this logic.